### PR TITLE
pm: misc cleanup and fixes

### PR DESCRIPTION
**Pull request by haukepetersen**
Follow up on #6160

This PR contains some collected cleanup and fixes for the PM, mostly some code re-organization, doxygen fixes and removal of un-used modules/includes.

### CODE REVIEW

**Comment by miri64** [comment]
Needs rebase

**Comment by haukepetersen** [comment]
rebased

**Comment by haukepetersen** [comment]
reabased again.

**Comment by haukepetersen** [comment]
@kaspar030 mind to take a look at this?

**Comment by haukepetersen** [comment]
@kaspar030 ping?!

Or @PeterKietzmann, @miri64, @kYc0o: would you guys mind to take a look?!

**Comment by miri64** [comment]
I'm fine with the code, but I don't have enough insight to test this at the moment.

File: drivers/include/periph/pm.h [hunk 1]
```diff
@@ -12,10 +12,7 @@
  * @brief       The kernels power management interface
  * @{
  *
- * The following functions *must* be available for every platform:
- *
- * pm_reboot()
- * pm_off()
+ * This interface *must* be implemented for every platform in RIOT.
```
Comment (PeterKietzmann) [inline drivers/include/periph/pm.h]: Is this already the case?

File: cpu/cortexm_common/Makefile.include [hunk 2]
```diff
@@ -1,2 +1,4 @@
 # include module specific includes
 export INCLUDES += -I$(RIOTCPU)/cortexm_common/include
+
+USEMODULE += cortexm_common_periph
```
Comment (PeterKietzmann) [inline cpu/cortexm_common/Makefile.include]: Shouln't be `USEMODULE += pm_layered` included here as well?

Comment (haukepetersen) [inline cpu/cortexm_common/Makefile.include]: Nope. The decision if the layered module should be used is up to a CPU. Some cortexm CPUs might want to use it (e.g. stms), but others might opt out (e.g. nrf5x)...

File: drivers/include/periph/pm.h (hunk 1 above)
Comment (haukepetersen) [inline drivers/include/periph/pm.h]: Yes, at least rudimentary.

**Comment by kYc0o** [comment]
It looks OK for me too and Murdock agreed, so I'd give my ACK.

File: drivers/include/periph/pm.h (hunk 1 above)
Comment (PeterKietzmann) [inline drivers/include/periph/pm.h]: I was wondering about `pm_off()` but before, I haven't seen there is fall-back code.

**Comment by kYc0o** [comment]
Why Jenkins shows red?

**Comment by PeterKietzmann** [comment]
So, I'm fine with this PR but I'd prefer @kaspar030 to hit the button. In worst case I will do, this afternoon.

**Comment by PeterKietzmann** [comment]
Just restarted the Jenkins build. Looks fine now.

File: cpu/msp430_common/periph/pm.c [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+
+void pm_reboot(void)
+{
+    /* force an hardware reboot ("Power-Up Clear"), by writing
```
Comment (kaspar030) [inline cpu/msp430_common/periph/pm.c]: I'm ok here, but in general, copy&pasting code doesn't remove copyright. :)

Comment (haukepetersen) [inline cpu/msp430_common/periph/pm.c]: Ups, this was not on purpose -> simply forgot to copy the old copyright into my template file... Sorry!

**Pull request merged**